docker/Dockerfile

```diff
@@ -0,0 +1,89 @@
+FROM debian:stretch as conpot-builder
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Add non-free packages, needed for snmp-mibs-downloader
+RUN sed -i -e 's/main/main non-free contrib/g' /etc/apt/sources.list
+
+# Install dependencies
+RUN apt-get update -y -qq && apt-get install --no-install-recommends -y -qq \
+        ipmitool \
+        supervisor \
+        tcpdump \
+        git \
+        python3-pip \
+        python3 \
+        python3-dev \
+        python3-setuptools \
+        build-essential \
+        python3-scapy \
+        python3-cffi \
+        wget &&\
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Copy the app from the host folder (probably a cloned repo) to the container
+RUN useradd -m conpot
+WORKDIR /opt/
+RUN git clone https://github.com/mushorg/conpot.git
+RUN chown conpot:conpot -R /opt/conpot
+
+# Install Python requirements
+USER conpot
+ENV PATH=$PATH:/home/conpot/.local/bin
+WORKDIR /opt/conpot
+RUN pip3 install --user --no-cache-dir coverage
+RUN pip3 install --user --no-cache-dir -r requirements.txt
+
+# Run test cases
+RUN py.test -v
+
+# Install the Conpot application
+RUN python3.5 setup.py install --user --prefix=
+
+WORKDIR /home/conpot
+RUN tar -zcf conpot.tgz .local
+
+FROM debian:stretch
+
+ENV DEBIAN_FRONTEND noninteractive
+USER root
+RUN sed -i -e 's/main/main non-free contrib/g' /etc/apt/sources.list
+
+WORKDIR /home/conpot
+
+RUN apt-get update -y -qq && apt-get install --no-install-recommends -y -qq \
+        ipmitool \
```

Comment:
Is there a specific reason for installing these dependencies twice?

Reply:
That is a good question as I did not explore if these are required or not, they are left over from original Dockerfile.

Reply:
About permission error, I always do docker-compose build to check if it builds before committing. This is weird, I will do full build to confirm.

Reply:
Sorry for not being clear about this.. :slightly_smiling_face: 

We are installing some deps (such as `wget`, `python3`, `ipmitool` etc.) twice:
First time here:
https://github.com/mushorg/conpot/blob/742b9584c0fd3c2b74cb1c75a0ac805cb4697876/docker/Dockerfile#L9-L23

Then again here:
https://github.com/mushorg/conpot/blob/742b9584c0fd3c2b74cb1c75a0ac805cb4697876/docker/Dockerfile#L55-L65

Would it be better to club them? Remove duplicates? :slightly_smiling_face:

Reply:
wget on first image is not needed but I do it twice as we build two images here first for building and second for running. Second image is clean so it needs some packages to be pulled second time.